functions/src/definitions/userHandlers.js

```diff
@@ -609,3 +649,43 @@ async function onTextListReceipt(messageObj, platform = "whatsapp") {
   await sendWhatsappTextMessage("user", from, response, null, true)
   return Promise.resolve(step)
 }
+
+async function createNewUser(userRef, messageTimestamp) {
+  const db = admin.firestore()
+
+  let success = false
+  let retries = 0
+
+  while (retries < 10 && !success) {
+    const referralId = nanoid(8)
+
+    success = await db.runTransaction(async (transaction) => {
+      const existingUserSnap = await transaction.get(
+        db.collection("users").where("referralId", "==", referralId).limit(1)
+      )
+
+      if (existingUserSnap.empty) {
+        // No user with this referralId exists, so add one
+        transaction.set(userRef, {
```

Comment:
I think `await` is missing here

Reply:
No need for await here, it's on line 663 at `await db.runTransaction`